src/FffGcodeWriter.cpp

```diff
@@ -1517,15 +1515,28 @@ bool FffGcodeWriter::processSkinPart(const SliceDataStorage& storage, LayerPlan&
         }
     }
 
-    int extra_infill_shift = 0;
-    coord_t offset_from_inner_skin_infill = 0;
+    Polygons skin_polygons;
+    Polygons skin_lines;
     Polygons concentric_perimeter_gaps; // the perimeter gaps of the insets of concentric skin pattern of this skin part
-    Polygons* perimeter_gaps_output = (fill_concentric_perimeter_gaps)? &concentric_perimeter_gaps : nullptr;
-    Infill infill_comp(pattern, skin_part.inner_infill, offset_from_inner_skin_infill, skin_line_width, skin_line_width, skin_overlap, skin_angle, z, extra_infill_shift, perimeter_gaps_output);
-    infill_comp.generate(skin_polygons, skin_lines);
+
+    const bool process_skin_infill = extruder_nr == top_bottom_extruder_nr;
```

Comment:
I don't understand what you mean.
How is this boolean an indirection?
Please comment in the final PR; your comments are already outdated.